### PR TITLE
Fix NPE in hash/equals code

### PR DESCRIPTION
**Pull request by chb0github**
Although there are JSR annotations insisting fields like image can't be null, they clearly aren't applied. Rather than delve into that, which isn't directly in the the way, this code simply handles the NPE case

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,33 +169,10 @@ Declarative agents can be defined like shown below. You can also reuse pre-confi
 of the template that you want to use as preconfigured. Only one label is expected to be specified.
 
 _Note_: You have to configure list of settings to be allowed in the declarative pipeline first (see the Allowed Overrides setting). They are disabled by default for security reasons, to avoid non-privileged users to suddenly be able to change certain settings.
-
 ## Usage
 
 The ECS agents can be used for any job and any type of job (Freestyle job, Maven job, Workflow job...), you just have to restrict the execution of the jobs on one of the labels used in the ECS Agent Template configuration. You can either restrict the job to run on a specific label only via the UI or directly in the pipeline.
 In addition, when configuring the cloud to run on, you must also
-
```
Comment (webratz) [inline README.md:177]: why removing the sample without inherit?

File: src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSTaskTemplate.java [hunk 2]
```diff
@@ -576,7 +576,7 @@ public String getDisplayName() {
     /**
      * This merge does not take an into consideration the child intentionally setting empty values for parameters like "entrypoint" - in fact
      * it's not uncommon to override the entrypoint of a container and set it to blank so you can use your own entrypoint as part of the command.
-     * What's really needed is a "MergeStrategy BinaryOperator ECSTaskTemplate that's user selectable.
+     * What's really needed is a "MergeStrategy `BinaryOperator<ECSTaskTemplate>` that's user selectable.
```
Comment (webratz) [inline src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSTaskTemplate.java:579]: it was that way before. the javadocs check on release failed due to that
you can check with ` mvn org.apache.maven.plugins:maven-javadoc-plugin:2.10.4:jar`

**Comment by webratz** [comment]
the build seems to have failed: https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Famazon-ecs-plugin/detail/PR-169/1/pipeline

File: src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSTaskTemplate.java (hunk 2 above)
Comment (chb0github) [reply src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSTaskTemplate.java:579]: Ok, strange. This is what I tried to merge in.

File: README.md (hunk 1 above)
Comment (chb0github) [reply README.md:177]: I think some things got lost in the merge let me double check things.

**Comment by chb0github** [comment]
Ok, yeah... @webratz  - So, here is the problem:

The NPE came from the fact that `image` was actually null during a hashCode phase even though it's labeled `@Nonnull`. So, I implemented the check in code - Maybe you can explain why the JSR annotations aren't being applied.

**Comment by webratz** [comment]
sadly not. i'm not much of a java dev, mostly working on it as we use this with my company.
I can try and have a look but might not be able to put in a lot of time in very soon

**Comment by webratz** [comment]
so to be clear: i'm fine with the null checks you added, but i think we should find a proper way to not break the usual plugin checks.
i also upgraded today all the dependencies from jenkins side, so maybe that also helps

**Comment by chb0github** [comment]
Ok, upon inspection, there are 2 different types of annotations here in use that look the same and aren't - And, as I don't understand object lifecycle in Jenkins I am loathe to get any further involved - annotations are tricky: They are meant "picked up" and so you can never be sure that a change won't break something.

However, I DO see that the 3 `Nonnull` values are constructor only (no setters) - so, I can actually remove the `@Nonnull` and simply check for it in the constructor. 

If you're not that savvy on this code and neither am I, then I need to proceed cautiously.

File: README.md (hunk 1 above)
Comment (chb0github) [reply README.md]: fixed

**Comment by chb0github** [comment]
Let me know if this now looks good and we can get another release -

**Comment by webratz** [comment]
released with 1.32

**Pull request merged**